MovieApp/MovieApp/MoviesViewController.m

```diff
@@ -37,51 +53,87 @@ - (void)viewDidLoad {
     self.navigationItem.titleView = searchBar;
     self.navigationItem.leftBarButtonItem = leftButton;
     self.navigationItem.rightBarButtonItem = rightButton;
-    
-    MovieDBDownloader *downloader = [MovieDBDownloader alloc];
-    [downloader configure];
-    [downloader getdMoviesByCriterion:TOP_RATED returnToHandler:self];
-    
 }
 
-
 - (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section{
-    return [movies count];
+    return movies ? [movies count] : 0;
     
 }
 
 - (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
     
-     MoviesCollectionViewCell *cell = [_moviesCollectionView dequeueReusableCellWithReuseIdentifier:[MoviesCollectionViewCell cellIdentifier] forIndexPath:indexPath];
+    MoviesCollectionViewCell *cell = [_moviesCollectionView dequeueReusableCellWithReuseIdentifier:[MoviesCollectionViewCell cellIdentifier] forIndexPath:indexPath];
     
-    cell.titleLabel.text=((Movie *)movies[indexPath.row]).title;
+    Movie *currentMovie=(Movie *)movies[indexPath.row];
     
-    return cell;
+    cell.titleLabel.text=currentMovie.title;
     
-}
+    NSDateFormatter *dateFormatter=[[NSDateFormatter alloc]init];
+    [dateFormatter setDateFormat:PREFFERED_DATE_FORMAT];
+    cell.releaseDateLabel.text=[dateFormatter stringFromDate:currentMovie.releaseDate];
+    
+    //hardcoded for now
+    cell.durationLabel.text = @"1h 45min";
+
+    cell.starsLabel.text=@"";
+    NSMutableString *stars=[[NSMutableString alloc]init];
+    NSUInteger numberOfStars=[self numberOfStarsFromRating:currentMovie.voteAverage];
```

Comment:
numberOfStars can be calculated inside cell that is responsible for presenting this info